tests/test_allowed_recipients.py

```diff
@@ -0,0 +1,714 @@
+        bokkyPooBahsDateTimeContract,
+    )
+    registry.addRecipient(recipient, "Test Recipient", {"from": manager})
+
+    top_up_factory = deployer.deploy(
+        TopUpAllowedRecipients, trusted_caller, registry, finance, ldo, easy_track
+    )
+
+    def make_call_data(recipients, amounts):
+        return encode_single("(address[],uint256[])", [recipients, amounts])
+
+    with reverts("EMPTY_DATA"):
+        top_up_factory.createEVMScript(trusted_caller, make_call_data([], []))
+
+    with reverts("LENGTH_MISMATCH"):
+        top_up_factory.createEVMScript(trusted_caller, make_call_data([recipient], []))
+
+    with reverts("LENGTH_MISMATCH"):
+        top_up_factory.createEVMScript(trusted_caller, make_call_data([], [123]))
+
+    with reverts("ZERO_AMOUNT"):
+        top_up_factory.createEVMScript(trusted_caller, make_call_data([recipient], [0]))
+
+    with reverts("ZERO_AMOUNT"):
+        top_up_factory.createEVMScript(
+            trusted_caller, make_call_data([recipient, recipient], [123, 0])
+        )
+
+    with reverts("RECIPIENT_NOT_ALLOWED"):
+        top_up_factory.createEVMScript(trusted_caller, make_call_data([stranger.address], [123]))
+
+
+def test_create_top_up_motion_above_limits(
+    entire_allowed_recipients_setup_with_two_recipients, agent
+):
+    (
+        easy_track,
+        _,  # evm_script_executor,
+        allowed_recipients_registry,
+        top_up_factory,
+        _,  # add_recipient_factory,
+        _,  # remove_recipient_factory,
+        recipient1,
+        _,  # recipient2,
+    ) = entire_allowed_recipients_setup_with_two_recipients
+
+    period_limit, period_duration = 100 * 10**18, 1
+    set_limit_parameters(period_limit, period_duration, allowed_recipients_registry, agent)
+
+    with reverts("SUM_EXCEEDS_SPENDABLE_BALANCE"):
+        create_top_up_motion([recipient1.address], [period_limit + 1], easy_track, top_up_factory)
+
+
+def test_limit_decreased_while_motion_is_in_flight(
+    entire_allowed_recipients_setup_with_two_recipients, agent
+):
+    (
+        easy_track,
+        _,  # evm_script_executor,
+        allowed_recipients_registry,
+        top_up_factory,
+        _,  # add_recipient_factory,
+        _,  # remove_recipient_factory,
+        recipient1,
+        _,  # recipient2,
+    ) = entire_allowed_recipients_setup_with_two_recipients
+
+    period_limit, period_duration = 100 * 10**18, 1
+    set_limit_parameters(period_limit, period_duration, allowed_recipients_registry, agent)
+
+    motion_id, motion_calldata = create_top_up_motion(
+        [recipient1.address], [period_limit], easy_track, top_up_factory
+    )
+
+    set_limit_parameters(period_limit // 2, period_duration, allowed_recipients_registry, agent)
+
+    chain.sleep(constants.MIN_MOTION_DURATION + 100)
+
+    with reverts("SUM_EXCEEDS_SPENDABLE_BALANCE"):
+        easy_track.enactMotion(
+            motion_id,
+            motion_calldata,
+            {"from": recipient1},
+        )
```

Comment:
Looks like we are lacking the case where we are creating two motions successfully but can enact only one because of the limits, or I'm missing something?

Reply:
I think this test fits the description. I'm right?
https://github.com/lidofinance/easy-track/blob/522ae893f6c03516354a8d1950b29b3203adae52/tests/test_allowed_recipients.py#L283

Reply:
Yep, it's fine, though. Looks like the one, I missed.